tests/test_driver.py

```diff
@@ -25,85 +26,58 @@
-    func_dir = xdg_config + "/fish/functions"
+def makeenv(script_path: Path, home: Path, test_helper_path: Path) -> dict[str, str]:
+    xdg_config = home / "xdg_config_home"
+    func_dir = xdg_config / "fish" / "functions"
     os.makedirs(func_dir)
-    os.makedirs(xdg_config + "/fish/conf.d/")
+    os.makedirs(xdg_config / "fish" / "conf.d")
     for func in (script_path / "test_functions").glob("*.fish"):
-        shutil.copy(func, func_dir + "/" + func.parts[-1])
+        shutil.copy(func, func_dir / func.parts[-1])
     shutil.copy(
-        script_path / "interactive.config", xdg_config + "/fish/conf.d/interactive.fish"
+        script_path / "interactive.config",
+        xdg_config / "fish" / "conf.d" / "interactive.fish",
     )
 
-    xdg_data = home + "/xdg_data_home"
+    xdg_data = home / "xdg_data_home"
     os.makedirs(xdg_data)
-    xdg_runtime = home + "/xdg_runtime_home"
+    xdg_runtime = home / "xdg_runtime_home"
     os.makedirs(xdg_runtime)
-    xdg_cache = home + "/xdg_cache_home"
+    xdg_cache = home / "xdg_cache_home"
     os.makedirs(xdg_cache)
-    tmp = home + "/temp"
+    tmp = home / "temp"
     os.makedirs(tmp)
 
-    # Compile fish_test_helper if necessary.
-    # If we're run multiple times, allow keeping this around to save time.
-    if test_helper_path:
-        thp = Path(test_helper_path)
-        if not os.path.exists(thp / "fish_test_helper"):
-            subprocess.run(
-                [
-                    "cc",
-                    script_path / "fish_test_helper.c",
-                    "-o",
-                    thp / "fish_test_helper",
-                ]
-            )
-        shutil.copy(thp / "fish_test_helper", home + "/fish_test_helper")
-    else:
-        subprocess.run(
-            [
-                "cc",
-                script_path / "fish_test_helper.c",
-                "-o",
-                home + "/fish_test_helper",
-            ]
-        )
-
-    # unset LANG, TERM, ...
-    for var in [
-        "XDG_DATA_DIRS",
-        "LANGUAGE",
-        "COLORTERM",
-        "KONSOLE_VERSION",
-        "TERM",  # Erase this since we still respect TERM=dumb etc.
-        "TERM_PROGRAM",
-        "TERM_PROGRAM_VERSION",
-        "VTE_VERSION",
-    ]:
-        if var in os.environ:
-            del os.environ[var]
-    langvars = [key for key in os.environ.keys() if key.startswith("LC_")]
-    for key in langvars:
-        del os.environ[key]
-
-    os.environ.update(
-        {
-            "HOME": home,
-            "TMPDIR": tmp,
-            "FISH_FAST_FAIL": "1",
-            "FISH_UNIT_TESTS_RUNNING": "1",
-            "XDG_CONFIG_HOME": xdg_config,
-            "XDG_DATA_HOME": xdg_data,
-            "XDG_RUNTIME_DIR": xdg_runtime,
-            "XDG_CACHE_HOME": xdg_cache,
-            "fish_test_helper": home + "/fish_test_helper",
-            "LANG": "C",
-            "LC_CTYPE": "en_US.UTF-8",
-        }
+    shutil.copy(test_helper_path, home / "fish_test_helper")
```

Comment:
Instead of copying, a hardlink would make more sense,
since it's effectively immutable, and we're on the same file system.

Actually, we shouldn't need to create a file at all,
since we could forward `test_helper_path`?
We don't currently run tests in a sandbox like bubblewrap so that should just work,
and we can always change it later.